node/relay_test.go

```diff
@@ -0,0 +1,73 @@
+package node
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	relayclient "github.com/libp2p/go-libp2p/p2p/protocol/circuitv2/client"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/log/logtest"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+)
+
+const relayConf = `{
+  "p2p":{
+    "listen": "/ip4/127.0.0.1/tcp/0",
+    "bootnodes": [],
+    "p2p-reachability": "public"
+  }
+}`
+
+func TestRelay(t *testing.T) {
+	configPath := filepath.Join(t.TempDir(), "config.json")
+	require.NoError(t, os.WriteFile(configPath, []byte(relayConf), 0o600))
+
+	relayAddrInfoCh = make(chan peer.AddrInfo)
+	t.Cleanup(func() { relayAddrInfoCh = nil })
+
+	cmd := GetCommand()
+	cmd.SetArgs([]string{"relay", "--config", configPath})
+	var eg errgroup.Group
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	eg.Go(func() error {
+		defer cancel()
+		return cmd.ExecuteContext(ctx)
+	})
+
+	var addrInfo peer.AddrInfo
+	select {
+	case addrInfo = <-relayAddrInfoCh:
+	case <-ctx.Done():
+		err := eg.Wait()
+		require.Error(t, err)
+		t.Fatalf("stopped with error: %v", err)
```

Comment:
NIT:

```suggestion
		require.FailNow("stopped with error:", err)
```

Reply:
In this case, my intent was that in case if the context is cancelled, `err` should not be nil
The cancellation is not expected here, but if it happens, I'd prefer to have this checked.
With the suggested change, `stopped with error: nil` doesn't necessarily indicate that `nil` error is also incorrect here

Reply:
I understand, but if we reach `require.Error` and it returns "expecting an error but got none" it would lead me down the wrong path in trying to fix it, but I'm not insisting on changing it 🙂

Reply:
Fixed